### PR TITLE
K8SPG-303 Scheduled backup e2e test

### PR DESCRIPTION
**Pull request by cap1984**
[K8SPG-303](https://jira.percona.com/browse/K8SPG-303) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=percona&utm_campaign=badge_info)

Making sure the backup schedule and multiple repos do work

**CHANGE DESCRIPTION**
---
**Problem:**
*No backup schedule process verification.*

**Cause:**
*Early v5 adoption.*

**Solution:**
*Adding the respective e2e test*

**CHECKLIST**
---
**Jira**
- [x] Is the Jira ticket created and referenced properly?
- [x] Does the Jira ticket have the proper statuses for documentation (`Needs Doc`) and QA (`Needs QA`)?
- [X] Does the Jira ticket link to the proper milestone (Fix Version field)?

**Tests**
- [ ] Is an E2E test/test case added for the new feature/change?
- [ ] Are unit tests added where appropriate?

**Config/Logging/Testability**
- [ ] Are all needed new/changed options added to default YAML files?
- [ ] Are the manifests (crd/bundle) regenerated if needed?
- [ ] Did we add proper logging messages for operator actions?
- [ ] Did we ensure compatibility with the previous version or cluster upgrade process?
- [ ] Does the change support oldest and newest supported PG version?
- [ ] Does the change support oldest and newest supported Kubernetes version?

### CODE REVIEW

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-417/bfe18bec1/init-deploy.log) |
| demand-backup | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-417/bfe18bec1/demand-backup.log) |
| scheduled-backup | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-417/bfe18bec1/scheduled-backup.log) |
| start-from-backup | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-417/bfe18bec1/start-from-backup.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-417/bfe18bec1/monitoring.log) |
| telemetry-transfer | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-417/bfe18bec1/telemetry-transfer.log) |
| We run 6 out of 6|

commit: https://github.com/percona/percona-postgresql-operator/pull/417/commits/bfe18bec1d61809eb58355c748843ce57176c850
image: `perconalab/percona-postgresql-operator:PR-417-bfe18bec1`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-417/17678b9a6/init-deploy.log) |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-417/17678b9a6/demand-backup.log) |
| scheduled-backup | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-417/17678b9a6/scheduled-backup.log) |
| start-from-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-417/17678b9a6/start-from-backup.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-417/17678b9a6/monitoring.log) |
| telemetry-transfer | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-417/17678b9a6/telemetry-transfer.log) |
| We run 6 out of 6|

commit: https://github.com/percona/percona-postgresql-operator/pull/417/commits/17678b9a67a6815ed88d9393764d6c956d7a7de8
image: `perconalab/percona-postgresql-operator:PR-417-17678b9a6`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-417/5647954a4/init-deploy.log) |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-417/5647954a4/demand-backup.log) |
| scheduled-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-417/5647954a4/scheduled-backup.log) |
| start-from-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-417/5647954a4/start-from-backup.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-417/5647954a4/monitoring.log) |
| telemetry-transfer | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-417/5647954a4/telemetry-transfer.log) |
| We run 6 out of 6|

commit: https://github.com/percona/percona-postgresql-operator/pull/417/commits/5647954a49e6087225161bb340a344f072df57a5
image: `perconalab/percona-postgresql-operator:PR-417-5647954a4`

File: e2e-tests/functions [hunk 1]
```diff
@@ -76,6 +88,17 @@ get_cr() {
 				mv "${TEMP_DIR}/start-from-backup.cr.yaml" "${TEMP_DIR}/cr.yaml"
 			fi
 			;;
+		scheduled-backup)
+			yq eval '.spec.backups.pgbackrest.configuration = [{"secret":{"name":"'${test_name}'-pgbackrest-secrets"}}]' "${TEMP_DIR}/cr.yaml" \
```
Comment (tplavcic) [inline e2e-tests/functions:92]: `yq` supports writing multiple commands in a single command, something like:
```
    yq eval '
        .metadata.name = "chaos-cluster-pod-kill" |
        del(.spec.selector.pods.test-namespace) |
        .spec.selector.pods.'$namespace'[0] = "'$pod'"' $conf_dir/chaos-pod-kill.yml \
        | kubectl apply -f -
```
might be a bit shorter and faster then piping

File: e2e-tests/tests/scheduled-backup/03-assert.yaml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+apiVersion: kuttl.dev/v1beta1
+kind: TestAssert
+timeout: 30
+---
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  name: 03-read-from-primary
+data:
+  data: ' 100500'
```
Comment (tplavcic) [inline e2e-tests/tests/scheduled-backup/03-assert.yaml:10]: Why empty space before number?

**Review by tplavcic — CHANGES_REQUESTED** [review]

File: e2e-tests/tests/scheduled-backup/03-assert.yaml (hunk 2 above)
Comment (cap1984) [reply e2e-tests/tests/scheduled-backup/03-assert.yaml:10]: yeah, it's a part of psql output

**Review by nmarukovich — DISMISSED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-417/e5139ed04/init-deploy.log) |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-417/e5139ed04/demand-backup.log) |
| scheduled-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-417/e5139ed04/scheduled-backup.log) |
| start-from-backup | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-417/e5139ed04/start-from-backup.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-417/e5139ed04/monitoring.log) |
| telemetry-transfer | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-417/e5139ed04/telemetry-transfer.log) |
| We run 6 out of 6|

commit: https://github.com/percona/percona-postgresql-operator/pull/417/commits/e5139ed04de63b4c3db68f95abfb82f6f38d3aee
image: `perconalab/percona-postgresql-operator:PR-417-e5139ed04`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-417/7f2908183/init-deploy.log) |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-417/7f2908183/demand-backup.log) |
| scheduled-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-417/7f2908183/scheduled-backup.log) |
| start-from-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-417/7f2908183/start-from-backup.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-417/7f2908183/monitoring.log) |
| telemetry-transfer | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-417/7f2908183/telemetry-transfer.log) |
| We run 6 out of 6|

commit: https://github.com/percona/percona-postgresql-operator/pull/417/commits/7f290818341c294fc6333af779be52cb446b6989
image: `perconalab/percona-postgresql-operator:PR-417-7f2908183`

**Review by tplavcic — APPROVED** [review]

**Review by nmarukovich — APPROVED** [review]

**Pull request merged**